src/dmd/cppmanglewin.d

```diff
@@ -684,22 +684,138 @@ private:
+                                    case "|":   ident = "?U";   goto continue_template;
+                                    default:    break;
+                                }
+                                break;
+                            case 2: // opOpAssign
+                                switch (str.peekSlice)
+                                {
+                                    case "/=":  ident = "?_0";  goto continue_template;
+                                    case "%=":  ident = "?_1";  goto continue_template;
+                                    case ">>=": ident = "?_2";  goto continue_template;
+                                    case "<<=": ident = "?_3";  goto continue_template;
+                                    case "&=":  ident = "?_4";  goto continue_template;
+                                    case "|=":  ident = "?_5";  goto continue_template;
+                                    case "^=":  ident = "?_6";  goto continue_template;
+                                    case "*=":  ident = "?X";   goto continue_template;
+                                    case "+=":  ident = "?Y";   goto continue_template;
+                                    case "-=":  ident = "?Z";   goto continue_template;
+                                    default:    break;
+                                }
+                                break;
+                            case 3: // opAssign
+                                ident = "?4";
+                                break;
+                            case 4: // opEquals
+                                ident = "?8";
+                                break;
+                            case 5: // opIndex
+                                ident = "?A";
+                                break;
+                            case 6: // opCall
+                                ident = "?R";
+                                break;
+                            default:
+                                break;
+                            continue_template:
+                                if (ti.tiargs.dim == 1)
+                                {
+                                    buf.writestring(ident);
+                                    return;
+                                }
+                                firstTemplateArg = 1;
+                                break;
+                        }
+                    }
+                }
+            }
+
             scope VisualCPPMangler tmp = new VisualCPPMangler((flags & IS_DMC) ? true : false);
             tmp.buf.writeByte('?');
             tmp.buf.writeByte('$');
-            auto id = ti.tempdecl.ident;
-            tmp.buf.writestring(id.toChars());
-            tmp.saved_idents[0] = id.toChars();
-            tmp.buf.writeByte('@');
+            tmp.buf.writestring(ident);
+            tmp.saved_idents[0] = ident;
+            if (ident == id.toChars)
```

Comment:
Please don't drop the () on the `toChars`

Reply:
Why?

Reply:
Because the `( )` are there on every other use in dmd. It's jarring to see it removed, and makes it look like a typo.
I prefer the paren-less versions only when using the . operator to chain function calls.

Reply:
Okay. I have had other PR push-backs because they wanted me to NOT match the surrounding code, and move slowly towards modern D style. It's hard to know the rules here ;)